mathics/core/atoms.py

```diff
@@ -561,6 +561,12 @@ def __hash__(self):
     def __str__(self) -> str:
         return base64.b64encode(self.value).decode("utf8")
 
+    # FIXME: the below does not use the "f" parameter to
+    # change behavior between FullForm and OutputForm
+    # Below we have the OutputForm behavior.
+    # A refactoring should be done so that this routine
+    # is removed and the form makes decisions, rather than
+    # have this routine know everything about all forms.
```

Comment:
Yep, I fully agree: `atom_to_boxes` should be removed in favor of formatting rules (`do_format_...`). However, in the meantime, I guess it is OK to implement here the right behavior, and then move the code to the right place.

Reply:
The problem here is that we really don't know all the cases and I don't want to find out right now and code in all of the possibilities. You haven't tried for example InputForm  or all of the other forms. 

Were this done this routine would be a mess. And when it is done the proper way it is organized by Form, not by atom it doesn't look like a mess. 

So again, I don't want to create a mess now to just have to clean that up later. My take is acknowledge the deficiency and move on.

Reply:
I am happy to revert all of the ByteArray stuff, and just leave the code that is needed to support NumPy 1.24.

Reply:
I just was talking about handling the basic cases. More general cases could be handled by adding rules, and eventually, removing the atom_to_boxes. But OK. Let's keep this simple and improve it later. In any case, I think that the best is that `__str__` returns the b64 encoded version of the bytearray.

Reply:
> I am happy to revert all of the ByteArray stuff, and just leave the code that is needed to support NumPy 1.24 as a _separate_ PR.

I do not think it is needed. Let's merge and continue.